portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,36 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> comments = new ArrayList<String>(); 
```

Comment:
To be technically correct, this isn't merely for historical reasons. It's a kind of foundational principle of "programming against interfaces instead of implementations".

The interface `List` exists so that you can program generically without consideration for idiosyncracies of the given list implementation. This is more important in some contexts than others. In this context, it's not particularly important, but the lesson I think is important.

Reply:
Oh, I did not know that. Thanks!